mergify_engine/tests/unit/test_command.py

```diff
@@ -54,3 +60,162 @@ def test_command_loader():
         "ignore_conflicts": True,
         "label_conflicts": "conflicts",
     }
+
+
+@pytest.mark.asyncio
+async def test_run_command_without_rerun_and_user(redis_cache):
+
+    client = mock.MagicMock()
+    client.auth.installation.__getitem__.return_value = 123
+    sub = subscription.Subscription(
+        redis_cache,
+        123,
+        True,
+        "",
+        {},
+        frozenset({}),
+    )
+
+    installation = context.Installation(123, "Mergifyio", sub, client, redis_cache)
+
+    repository = context.Repository(installation, "demo", 123)
+
+    ctxt = await context.Context.create(
+        repository,
+        {
+            # https://docs.github.com/en/rest/reference/repos#collaborators
+            "number": 789,
+            "state": "open",
+            "title": "Amazing new feature",
+            "user": {
+                "login": "octocat",
+                "id": 1,
+            },
+            "mergeable_state": "ok",
+            "merged_by": None,
+            "merged": None,
+            "merged_at": None,
+            "base": {
+                "sha": "sha",
+                "ref": "main",
+                "user": {
+                    "login": {
+                        "Mergifyio",
+                    },
+                },
+                "repo": {
+                    "name": "demo",
+                    "private": False,
+                    "owner": {
+                        "login": "Mergifyio",
+                        "id": 123,
+                    },
+                    "permissions": {
+                        "admin": False,
+                        "push": False,
+                        "pull": True,
+                    },
+                },
+            },
+            "requested_teams": [
+                {
+                    "id": 1,
+                    "name": "Justice League",
+                    "slug": "justice-league",
+                    "description": "A great team.",
+                    "privacy": "closed",
+                    "permission": "admin",
+                }
+            ],
+        },
+    )
+
+    with pytest.raises(RuntimeError) as error_msg:
+        await handle(ctxt=ctxt, comment="@Mergifyio update", user=None)
+    assert "user must be set if rerun is false" in str(error_msg.value)
+
+
+@pytest.mark.asyncio
+class Test_run_command_without_permissions(TestCase):
```

Comment:
Why do you have a class here ?

Reply:
I would like to check the logging message generated by `log(message)`. I saw I can use `assertLogs()` from TestCase.

Reply:
https://docs.pytest.org/en/stable/logging.html